src/main/java/io/github/oliviercailloux/y2018/StudentsPreference.java

```diff
@@ -1,27 +1,80 @@
 package io.github.oliviercailloux.y2018;
 
+
+import java.util.Arrays;
 import java.util.HashMap;
+import java.util.HashSet;
 import java.util.Map;
 import java.util.Set;
 
-public class StudentsPreference {
+import javax.json.bind.Jsonb;
+
+
+import com.google.common.base.Strings;
 
+
+
+public class StudentsPreference {
+	
 	private Map<Person, Set<RawPreference>> studPref;
+	
+	private static Jsonb jsonb = JsonUtils.getInstance();
 
 	public StudentsPreference() {
-		studPref = new HashMap<>();
+		studPref = null;
+	}
+	
+	/**
+	 * CONSTRUCTOR
+	 *
+	 * @param student
+	 *            not <code>null</code>.
+	 * @param les_preferences
+	 *                 
+	 */
+	public StudentsPreference(Person student, Set<RawPreference> les_preferences) {
+		this.studPref = new HashMap<Person, Set<RawPreference>>();
+		this.studPref.put(student, les_preferences);
+				
 	}
 
 	public Map<Person, Set<RawPreference>> getStudPref() {
 		return this.studPref;
 	}
-
+	
 	public void addPref(Person student, RawPreference pref) {
 		studPref.get(student).add(pref);
 	}
-
+	/**
+	 * @param student not <code>null</code>.
+	 * @return not <code>null</code>.
+	 * */
 	public Set<RawPreference> getPreference(Person student) {
 		return studPref.get(student);
 	}
+	
+	/**
+	 * this method has been added to transform StudentsPreference into json
+	 * the order is inherited from the order of RawPreference 
+	 * */
+	public String studentsPreferenceToJson() {
+		String result ="";
+		for(Person person : this.studPref.keySet()) {
+			result+=person.toJson();
```

Comment:
Inélégant et fragile. Utiliser JSON-B ou JSON-P.

Reply:
D’ailleurs, illustration : votre résultat n’est pas du JSON valide.

Reply:
Monsieur j'ai une question sur cette partie, puisque j'ai essayé jsonb.toJson(this); dans la méthode mais cela ne fonctionne pas (j'ai "wrong number of arguments" comme erreur)
je n'arrive pas à résoudre ce problème.

merci,